File: Practice07/ListView02/app/src/main/java/com/example/listview2/MainActivity.java
package com.example.listview2;

import android.os.Bundle;
import android.app.Activity;
import android.view.View;
import android.view.View.*;
import android.content.Intent;
import android.widget.*;
public class MainActivity extends Activity{
    Button jbn;
    Bundle bdl;
    Intent itn;
    public void onCreate(Bundle b) {
        super.onCreate(b);
        setContentView(R.layout.activity_main);
        jbn = (Button) findViewById(R.id.xbn);
        jbn.setOnClickListener(new OnClickListener() {
            public void onClick(View arg0) {
                itn = new Intent(MainActivity.this, SegundaActivity.class);
                bdl = new Bundle();
                itn.putExtras(bdl);
                startActivity(itn);
            }
        });
    }
}

File: Practice25/Kotlin02/app/src/main/java/com/example/kotlin02/MainActivity.kt
package com.example.kotlin02

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //Operador de llamada segura
        val string: String? = "Hello World!"
        print(string?.length)  // Will print the string's length, or "null" if the string is null.

        //Operador Elvis
        var a: String? = "Nullable String Value"
        val b = a?.length ?: -1

        //Operador !!
        //val x: Int? = null
        //val y = x!!.toDouble()

        //Expresión lambada
        btn1.setOnClickListener {view -> Toast.makeText(this, "Button clicked", Toast.LENGTH_SHORT).show() }

        //Funciones de extenseión
        var multiplicacion = 20.multiplicar(5)
        println("El resultado de la multiplicacion es $multiplicacion == ${20.times(5)}")


    }
    fun Int.multiplicar(numero: Int): Int {
        return this * numero
    }
}
